lib/ProductOpener/Ecoscore.pm

```diff
@@ -842,6 +853,19 @@ sub compute_ecoscore ($product_ref) {
 			add_tag($product_ref, "misc", "en:ecoscore-not-computed");
 		}
 	}
+
+	# Track if ecoscore has changed through different Agribalyse versions
+	if (defined $old_ecoscore_score && $old_ecoscore_score != $product_ref->{ecoscore_score}) {
+		$product_ref->{ecoscore_data}{previous_data} = {
+			grade => $old_ecoscore_grade,
+			score => $old_ecoscore_score,
+			version => $old_ecoscore_version // "3.0"
+		};
+		add_tag($product_ref, "misc", "en:ecoscore-changed");
```

Comment:
the misc_tags are not persistent, we recompute them after each product update, so if we do  the add_tag here, it will be executed once, and it will be removed at the next product update.

Maybe we could move the add_tag in another check where we compare $product_ref->{ecoscore_data}{previous_data}[score} to $product_ref->{ecoscore_score}

Reply:
OK, makes sense

Reply:
I just tried to create a failing test for this but my new tags are retained, the code currently only removes specific tags. Is there some other process that would remove all of the tags?

Reply:
I was looking into it, it looks like we reset misc_tags when we compute the Nutri-Score. So if you use the update_all_products.pl script to compute the Eco-Score, they may not be removed.

The misc_tags things is very messy, I added it as a temporary debug thing and then it grew into something else..

Reply:
OK. I don't really want to change other code. I'm guessing it will be easy enough to do a query without the tags present anyway?